src/Microsoft.Extensions.DependencyInjection/ServiceLookup/ClosedIEnumerableService.cs

```diff
@@ -3,19 +3,18 @@
 
 using System;
 using System.Collections.Generic;
-using System.Linq;
 
 namespace Microsoft.Extensions.DependencyInjection.ServiceLookup
 {
     internal class ClosedIEnumerableService : IService
     {
-        private readonly Type _itemType;
         private readonly ServiceEntry _serviceEntry;
 
         public ClosedIEnumerableService(Type itemType, ServiceEntry entry)
         {
-            _itemType = itemType;
+            ServiceType = itemType;
             _serviceEntry = entry;
+            ImplementationType = itemType.MakeArrayType();
```

Comment:
This happens to be an array today, but that's really an implementation detail. It could be changed to any IEnumerable implementation. Since this is only used to see if IDisposable is implemented, why not just make it null like we do for factories?